google/cloud/iot/v1/resources.proto

```diff
@@ -193,14 +197,21 @@ message EventNotificationConfig {
 
   // A Cloud Pub/Sub topic name. For example,
   // `projects/myProject/topics/deviceEvents`.
-  string pubsub_topic_name = 1;
+  string pubsub_topic_name = 1 [
+    // TODO: This requires the client library to have a dependency on pubsub; is this acceptable?
+    // Note that it doesn't require this proto to have a dependency on the pubsub proto.
+    (google.api.resource_type) = "google.pubsub.v1.Topic" 
+  ];
 }
 
 // The configuration for notification of new states received from the device.
 message StateNotificationConfig {
   // A Cloud Pub/Sub topic name. For example,
   // `projects/myProject/topics/deviceEvents`.
-  string pubsub_topic_name = 1;
+  string pubsub_topic_name = 1 [
+    // TODO: See comment for EventNotificationConfig.pubsub_topic_name
+    (google.api.resource_type) = "google.pubsub.v1.Topic" 
```

Comment:
Oooooh this is a challenge actually. It is unlikely that google.pubsub.v1.Topic would be imported. :-/

Reply:
It doesn't need importing at the proto level. That is, this file doesn't need an "import google.pubsub.v1"
But a dependency on the client library *will* be required. This will only affect languages that implement resource-names (so only Java + C# at the moment).

Reply:
Oops, I'm completely wrong here. It certainly does need importing in the proto, otherwise the generator can't access the `resource` or `resource_set`.
Therefore I guess this is untenable. API producers won't want these extra dependencies, especially as it won't be at all obvious what they are for.

Reply:
Isn't there a semantic dependency on pubsub.Topic anyway? I can't speak to the whims of API producers, but implied dependencies IMO should be able to be made explicit if possible.